### DISCUSSION TITLE
VSIX Project (Community) - build errors on empty project `error MSB4801` `error CS0246`

### DISCUSSION THREAD

**Discussion by cjdutoit** [VsixCommunity/Community.VisualStudio.Toolkit#404] [Q&A] [answered]
Hi all,

I am on the latest version of VS2022 (17.4.3) and are currently seeing build errors on GitHub on a brand new VSIX project without any code in it.

Building the project within Visual Studio is fine, but doing a build on GitHub for PR's fails.  I can replicate this locally by running `dotnet build` on the command line.

I get the following error on a few files:
`.nuget\packages\microsoft.vssdk.buildtools\17.0.5232\build\Microsoft.VSSDK.BuildTools.targets(16,5): error MSB4801: The task factory "CodeTaskFactory" is not supported on the .NET Core version of MSBuild. [C:\repos\VSIXProject3\VSIXProject3\VSIXProject3.csproj]`

`.nuget\packages\community.visualstudio.toolkit.17\17.0.430\build\imports\NewtonsoftJsonVersionCheck.targets(38,5): warning CVSTBLD002: The version of Newtonsoft.Json cannot be greater than 13.0.1, but version  is currently being used. [C:\repos\VSIXProject3\VSIXProject3\VSIXProject3.csproj]`

I then tried the non-community VSIX projects and seems to be getting errors there as well on the same files although the error code is now different

`error CS0246: The type or namespace name 'UseManagedResourcesOnly' could not be found (are you missing a using directive or an assembly reference?) [C:\repos\VSIXProject1\VSIXProject1\VSIXProject1\VSIXProject1.csproj]`

Have any of you seen?  Any suggestions on a fix?

Thanks
Chris

**Comment by reduckted** [comment, marked as answer]
Unfortunately you cannot (currently) use the .NET CLI to compile a Visual Studio extension.

https://developercommunity.visualstudio.com/t/VSIX-project-with-SDK-style-csproj/1572145

Thankfully it appears to be in progress (see [this comment](https://developercommunity.visualstudio.com/t/VSIX-project-with-SDK-style-csproj/1572145#T-N10138582) on that ticket and the [What's Next](https://devblogs.microsoft.com/visualstudio/visual-studio-2022-17-4/#whats-next-visual-studio-2022-17-5-preview-1) section in the blog post about VS 17.4). So hopefully in 17.5 we'll be able to use the .NET CLI 🤞 

In the meantime, you can still build the extension in a GitHub action, but you have to use MSBuild to do it via the [`microsoft/setup-msbuild`](https://github.com/marketplace/actions/setup-msbuild) action. Here's an example: 
https://github.com/reduckted/ProjectFilter/blob/19950f9ab2add92964a83a2052b29a9852be4bad/.github/workflows/ci.yml#L16-L33

**Comment by cjdutoit** [comment]
@reduckted Thanks for the feedback and the pointer to use msbuild.  I will update my GitHub action. Appreciate your help.
